akka-typed/src/main/scala/akka/typed/Props.scala

```diff
@@ -3,18 +3,23 @@
  */
 package akka.typed
 
-import akka.actor.Deploy
-import akka.routing.RouterConfig
+import java.util.concurrent.Executor
+import scala.concurrent.ExecutionContext
+
+sealed trait DispatcherSelector
+case object DispatcherDefault extends DispatcherSelector
+case class DispatcherFromConfig(path: String) extends DispatcherSelector
+case class DispatcherFromExecutor(executor: Executor) extends DispatcherSelector
+case class DispatcherFromExecutionContext(ec: ExecutionContext) extends DispatcherSelector
 
 /**
  * Props describe how to dress up a [[Behavior]] so that it can become an Actor.
  */
-@SerialVersionUID(1L)
-final case class Props[T](creator: () ⇒ Behavior[T], deploy: Deploy) {
-  def withDispatcher(d: String) = copy(deploy = deploy.copy(dispatcher = d))
-  def withMailbox(m: String) = copy(deploy = deploy.copy(mailbox = m))
-  def withRouter(r: RouterConfig) = copy(deploy = deploy.copy(routerConfig = r))
-  def withDeploy(d: Deploy) = copy(deploy = d)
+final case class Props[T](creator: () ⇒ Behavior[T], dispatcher: DispatcherSelector, queueSize: Int) {
```

Comment:
is `queueSize` max mailbox queue size? Then I think the word `max` should be included.